### PR TITLE
FOGL-1900 : SP/C++ : investigation on the stability

### PR DESCRIPTION
**Pull request by sstefano69**
Code ready : 
* final testing

note : 
* SOLVED - the execution freeze in some runs using a previous version, investigating... different situation respect the previous 'segfault'

### CODE REVIEW

File: C/common/storage_client.cpp [hunk 1]
```diff
@@ -27,19 +29,75 @@ using HttpClient = SimpleWeb::Client<SimpleWeb::HTTP>;
...
+ * stores the provided HttpClient into the map
+ */
+StorageClient::StorageClient(HttpClient *client) {
+
+	std::thread::id thread_id = std::this_thread::get_id();
+
+	// FIXME:
+	m_logger->debug("DBG - StorageClient 2 - thread_id -%x-", thread_id);
+
+	m_client_map[thread_id] = client;
```
Comment (MarkRiddoch) [inline C/common/storage_client.cpp]: We should protect access to the map using a mutex as any thread may write this.

File: C/common/storage_client.cpp [hunk 2]
```diff
@@ -27,19 +29,75 @@ using HttpClient = SimpleWeb::Client<SimpleWeb::HTTP>;
...
+
+	std::map<std::thread::id, HttpClient *>::iterator item;
+
+	HttpClient *client;
+
+	std::thread::id thread_id = std::this_thread::get_id();
+
+	m_logger->debug("DBG 9 - thread_id -%x-", thread_id);
+
+	item = m_client_map.find(thread_id);
```
Comment (MarkRiddoch) [inline C/common/storage_client.cpp]: Again, I would protect this map access within a mutex

**Comment by sstefano69** [comment]
@AmandeepArora, please try this code, it should close to the finale one. thanks.

**Comment by AmandeepArora** [comment]
@sstefano69 - Can you merge the latest code from develop into this branch? I'll test using your branch tomorrow morning.

**Comment by AmandeepArora** [comment]
Sending process is working stably for me. But I need this FOGL-1900 branch for some more time while I complete the benchmarking activity. So don't delete this branch today.

**Pull request merged**